src/pages/ManageComputeHost/DeleteComputeHost.js

```diff
@@ -14,9 +14,143 @@
 **/
 
 import DeleteCompute from '../ReplaceServer/DeleteCompute.js';
+import { translate } from '../../localization/localize.js';
+import * as constants from '../../utils/constants.js';
+import { deleteJson, putJson } from '../../utils/RestUtils.js';
+import { logProgressError, logProgressResponse } from '../../utils/MiscUtils.js';
 
 class DeleteComputeHost extends DeleteCompute {
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      ...this.state,
+    };
+  }
+
+  getSteps() {
+    // Added steps that are necessary before delete
+    let steps = [{
+      label: translate('server.deploy.progress.remove_from_aggregates'),
+      playbooks: [constants.REMOVE_FROM_AGGREGATES_ACTION]
+    }, {
+      label: translate('server.deploy.progress.disable_network_agents'),
+      playbooks: [constants.DISABLE_NETWORK_AGENTS_ACTION]
+    }];
+
+    // Append parent's steps
+    let parentSteps = super.getSteps();
+    steps = steps.concat(parentSteps);
+
+    return steps;
+  }
+
+  getPlaybooks() {
+    // Added playbooks that are necessary before delete
+    let playbooks = [{
+      name: constants.REMOVE_FROM_AGGREGATES_ACTION,
+      action: ((logger) => {
+        return this.removeAggregates(logger);
+      })
+    }, {
+      name: constants.DISABLE_NETWORK_AGENTS_ACTION,
+      action: ((logger) => {
+        return this.disableNetworkAgents(logger);
+      })
+    }];
+
+    // Append parent's playbooks
+    let parentBooks = super.getPlaybooks();
+    playbooks = playbooks.concat(parentBooks);
+
+    return playbooks;
+  }
+
+  removeAggregates = (logger) => {
+    const apiUrl =
+      '/api/v2/compute/aggregates/' + this.props.operationProps.oldServer.hostname;
+    logger('DELETE ' + apiUrl);
+    return deleteJson(apiUrl)
+      .then((response) => {
+        const logMsg =
+          'Got response from removing aggregates for compute host ' +
+          this.props.operationProps.oldServer.hostname;
+        logProgressResponse(logger, response, logMsg);
+      })
+      .catch((error) => {
```

Comment:
This error handling logic is pretty convoluted, and seems to be copy/pasted below. It's unclear why a failure dialog is only presented for some 500 errors and not others. Or why some messages are translated and others are not, or why some errors cause an error to be thrown and others do not.

Reply:
@GarySmith  it is copied and pasted. I tried to move them to a common place like util, but gave it up , the effort seems to be more involved than I initially thought. 
When delete aggregates, since the backend deletes host aggregates one by one, if there are errors, we don't stop, in the end if the results include good and bad responses, backends returns response with 500 status code. Then UI checks if error code is 500 and have failed (bad) and have deleted(good), pops a confirm dialog to ask user to continue or not.

Reply:
Talked to Gary, I will add some comments so we will be a little bit clear about the error handling part.